pkg/controllers/provisioningv2/rke2/bootstrap/controller.go

```diff
@@ -33,27 +35,30 @@ const (
 )
 
 type handler struct {
-	serviceAccountCache corecontrollers.ServiceAccountCache
-	secretCache         corecontrollers.SecretCache
-	secretClient        corecontrollers.SecretClient
-	machineCache        capicontrollers.MachineCache
-	capiClusters        capicontrollers.ClusterCache
-	deploymentCache     appcontrollers.DeploymentCache
-	rkeControlPlanes    rkecontroller.RKEControlPlaneCache
-	rkeBootstrapClient  rkecontroller.RKEBootstrapClient
+	serviceAccountCache     corecontrollers.ServiceAccountCache
+	secretCache             corecontrollers.SecretCache
+	secretClient            corecontrollers.SecretClient
+	machineCache            capicontrollers.MachineCache
+	capiClusters            capicontrollers.ClusterCache
+	deploymentCache         appcontrollers.DeploymentCache
+	rkeControlPlanes        rkecontroller.RKEControlPlaneCache
+	rkeProvisioningClusters v1.ClusterClient
```

Comment:
```suggestion
	rkeClusters provcontrollers.ClusterClient
```

Reply:
Do we need the client here or will the cache suffice?

Reply:
I tried to use a cache but it was always missing the cluster object, so a client seemed like the only option. If you have any tips on why that might be I'm all ears, but using `clients.Provisioning.Cluster().Cache()` seemed to be updating fast enough for us to use it here.